src/plugin/buddhistEra/index.js

```diff
@@ -6,14 +6,15 @@ export default (o, c) => { // locale needed later
   // extend en locale here
   proto.format = function (formatStr, localeObject) {
     const yearBias = 543
+    const utils = this.$utils()
     const locale = localeObject || this.$locale()
     const str = formatStr || FORMAT_DEFAULT
     const result = str.replace(/BBBB|BB/g, (match) => {
       switch (match) {
         case 'BB':
-          return (this.$y + yearBias) % 100
+          return utils.padStart(String((this.$y + yearBias) % 100), 2, '0')
         default: // BBBB
-          return this.$y + yearBias
+          return utils.padStart(String(this.$y + yearBias), 4, '0')
```

Comment:
@saknarak we don't need this in four-digits year right? e.g. year 998 should be 998 not 0998

and also come up with a new idea, `String(this.$y + yearBias).slice(-2)` seems easier to understand.

Reply:
base on moment
console.log(moment('0123-05-07').format('DD MMM YYYY')) // 07 MAY 0123
console.log(moment('123-05-07').format('DD MMM YYYY')) // error

Reply:
I see